src/bcr_api/bwresources.py

```diff
@@ -1155,7 +1179,93 @@ def upload_all(
             if name in self.ids:
                 cat_data[name] = self.ids[name]
         return cat_data
+    
+    def upload_rule_categories(
+        self, create_only=False, modify_only=False, overwrite_children=False, **kwargs
+    ):
+        """
+        Uploads a category.
+        You can upload a new category, add subcategories to an existing category, overwrite the subcategories of an existing category, or change the name of an existing category with this function.
+        Args:
+            create_only:        If True and the category already exists, no action will be triggered - Optional.  Defaults to False.
+            modify_only:        If True and the category does not exist, no action will be triggered - Optional.  Defaults to False.
+            overwrite_children: Boolen flag that indicates if existing subcategories should be appended or overwriten - Optional.  Defaults to False (appending new subcategories).
+            kwargs:             You must pass in name (parent category name) and children (dict of subcategories).  You can optionally pass in multiple (boolean - indicates if subcategories are mutually exclusive) and/or new_name (string) if you would like to change the name of an existing category.
+        Returns:
+            A dictionary of the form {id: categoryid, multiple: True/False, children: {child1name: child1id, ...}}
+        """
+        data_list = [kwargs]
+        for data in data_list:
+            if "name" not in data:
+                raise KeyError("Need name to upload " + self.parameter, data)
+            elif not data.get("children"):
+                raise KeyError("Need children to upload categories", data)
+            else:
+                name = data["name"]
+
+            validated_children = []
+            for child in data["children"]:
+                rules = []
+                if child.get("rules", []):
+                    for rule in child["rules"]:
+                        rules.append(self._fill_subrule_data(rule))
+                child["rules"] = rules
+                validated_children.append(child)
+            data["children"] = validated_children
```

Comment:
I actually went ahead and did some tests on this. Currently when trying to go through these steps, you don't seem to  experience the problem you described (losing valuable markup). Instead, you just get an error telling you that the category already exists, which I think is less of a problem. 

```
def create_test_resources():    
    assert isinstance(categories, BWCategories)
    assert isinstance(mentions, BWMentions)
    assert hasattr(categories, 'upload_rule_categories')
    
    # delete the test parent category if it exists
    if test_parent_category_name in [i for i in categories.ids]:
        print(f'deleting test parent category ({test_parent_category_name})')
        categories.delete(test_parent_category_name)
    
    # upload a parent category with a child category. We don't need to use upload_rule_categories() for this, as it can be just a regular category with no rules associated
    categories.upload(name = test_parent_category_name, 
                  children = [test_child_category_name])
                      
    # delete test query if it exists
    if test_query_name in queries.names.values():
        print(f'deleting test query ({test_query_name})')
        queries.delete(test_query_name)
     
     # upload test query            
    queries.upload(name=test_query_name, startDate=yesterday_bcr_format, booleanQuery='ice cream', languages=['en'], samplePercentage=1,)
    
    
def check_manual_markup_on_category_not_lost_with_upload_rule_categories(test_query_name):
                      
    # patch some mentions into the child category, setting max_pages to be 1, so we only collect a few mentions
    some_mentions = queries.get_mentions(name=test_query_name, startDate=yesterday_bcr_format,max_pages=1)
    
    mentions_to_patch = random.sample(some_mentions, 10)
    print(f'number of mentions: {len(some_mentions)}')
    
    mentions.patch_mentions(mentions_to_patch, action = "addCategories", setting = {test_parent_category_name:[test_child_category_name]})
    
    
    # add a new child category to the parent category
    
    categories.upload_rule_categories(name=test_parent_category_name, children=example_children)
    
    print(categories.get(test_parent_category_name))
    
    return categories.get(test_parent_category_name)
```

Running `check_manual_markup_on_category_not_lost_with_upload_rule_categories(test_parent_category_name)` raises the error:

> RuntimeError: [{'code': 201, 'message': 'Parent category with that name already exists'}]

So I'm not sure it's a problem, really. If someone wants to add a subcategory to an existing parent category, they can just use `categories.get() ` to get the parent and child category names, boolean, and filters, delete, and reupload.